contracts/buy/PartyBuy.sol

```diff
@@ -0,0 +1,180 @@
+        // set storage variables
+        nftContract = IERC721Metadata(_nftContract);
+        tokenId = _tokenId;
+        name = _name;
+        symbol = _symbol;
+        expiresAt = _secondsToTimeout + block.timestamp;
+        require(_maxPrice > 0, "PartyBuy::initialize: must set price higher than 0");
+        maxPrice = _maxPrice;
+        // validate that party split won't retain the total token supply
+        uint256 _remainingBasisPoints = 10000 - TOKEN_FEE_BASIS_POINTS;
+        require(_splitBasisPoints < _remainingBasisPoints, "PartyBuy::initialize: basis points can't take 100%");
+        // validate that a portion of the token supply is not being burned
+        if (_splitRecipient == address(0)) {
+            require(_splitBasisPoints == 0, "PartyBuy::initialize: can't send tokens to burn addr");
+        }
+        splitBasisPoints = _splitBasisPoints;
+        splitRecipient = _splitRecipient;
+        // validate token exists
+        require(_getOwner() != address(0), "PartyBuy::initialize: NFT getOwner failed");
+    }
+
+    // ======== External: Contribute =========
+
+    /**
+     * @notice Contribute to the Party's treasury
+     * while the Party is still active
+     * @dev Emits a Contributed event upon success; callable by anyone
+     */
+    function contribute() external payable nonReentrant {
+        // require that the new total contributed is not greater than
+        // the maximum amount the Party is willing to spend
+        require(totalContributedToParty + msg.value <= getMaximumContributions(), "PartyBuy::contribute: cannot contribute more than max");
+        // continue with shared _contribute flow
+        _contribute();
+    }
+
+    // ======== External: Buy =========
+
+    /**
+     * @notice Buy the token by calling targetContract with calldata supplying value
+     * @dev Emits a Bought event upon success; reverts otherwise. callable by anyone
+     */
+    function buy(uint256 _value, address _targetContract, bytes calldata _calldata) external nonReentrant {
+        require(
+            partyStatus == PartyStatus.ACTIVE,
+            "PartyBuy::buy: party not active"
+        );
+        // check that value is not more than the maximum price set at deploy time
+        require(_value <= maxPrice, "PartyBuy::buy: can't spend over max price");
+        // check that value is not more than
+        // the maximum amount the party can spend while paying ETH fee
+        require(_value <= getMaximumSpend(), "PartyBuy::buy: insuffucient funds to buy token plus fee");
+        // execute the calldata on the target contract
+        address(_targetContract).call{value: _value}(_calldata);
+        // NOTE: we don't are if the call succeeded
+        // as long as the NFT is owned by the Party
+        require(_getOwner() == address(this), "PartyBuy::buy: failed to buy token");
+        // set partyStatus to WON
+        partyStatus = PartyStatus.WON;
+        // record totalSpent,
```

Comment:
would prefer for these comments to be in `_closeSuccessfulParty` instead of here.  seeing them here makes me worried code is missing to achieve these items

Reply:
they are in `_closeSuccessfulParty`, but duplicated here to make clear all that's going on under the hood in this call. i was concerned because this call writes to storage in several places, and a careless developer could get confused using it (which is usually a sign of not the best / most extensible code, but in this case, it is nice to have a smaller surface area to review by removing duplicated code)